src/runtime/refiner.ts

```diff
@@ -1237,143 +1241,220 @@ export class Fraction {
   }
 }
 
-export class Polynomial {
-  private _coeffs: number[];
-  indeterminate?: string;
+export class Term {
+  private _indeterminates: Dictionary<number>;
+
+  constructor(indeterminates: Dictionary<number> = {}) {
+    this.indeterminates = indeterminates;
+  }
+
+  static copyOf(tm: Term): Term {
+    return new Term(tm.indeterminates);
+  }
+
+  get indeterminates(): Dictionary<number> {
+    for (const [indeterminate, power] of Object.entries(this._indeterminates)) {
+      if (power === 0) {
+        delete this._indeterminates[indeterminate];
+      }
+    }
+    const ordered = {};
+    const unordered = this._indeterminates;
+    Object.keys(unordered).sort().forEach((key) => {
+      ordered[key] = unordered[key];
+    });
+    this._indeterminates = ordered;
```

Comment:
I think we need to explain why we're doing this (as its somewhat complicated and looks a bit like a noop).

Reply:
Also, do we need to sort the indeterminates if we stringify the object for the map?